docs/topics/patterns/instant-messaging.mdx

```diff
@@ -0,0 +1,359 @@
+---
+sidebar_position: 10
+title: Instant Messaging Pattern
+sidebar_label: Instant messaging
+description: Learn best practices for building instant messaging with Momento Cache and Topics.
+keywords:
+ - topics
+ - development patterns
+ - momento
+ - chat
+ - instant messaging
+ - enterprise patterns
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Use Momento Cache and Topics to build instant messaging
+
+Instant messaging is a must-have feature in many applications. Communicating seamlessly in one-to-one, one-to-many, and many-to-many conversations is often thought of as a necessary component of an application. With Momento Cache and Topics, you can build instant messaging into your application without any backend infrastructure. Using the [Momento web SDK](../../sdks/web), you can access your cache data and publish and subscribe to messages directly in your browser.
+
+:::info
+This pattern uses **Momento Cache** to store historical messages and **Momento Topics** to facilitate real-time communication between instant messaging participants.
+:::
+
+## Architecture
+
+Building instant messaging on Momento can be done entirely client-side, meaning in your user interface. Browsers, iPhone apps, and Android apps all have the capability to access resources in Momento directly.
+
+![Diagram of client applications communicating directly via Momento](@site/static/img/topics/patterns/instant-messaging-architecture.png)
+
+In the diagram above, messages are being loaded from a [sorted set](../../cache/develop/basics/datatypes#sorted-sets) when a new user joins the chat. Users subscribe to a chatroom-specific topic to recieve messages in real-time while they are active in the chat. As messages are sent, they are added directly to the sorted set by the sender.
+
+## Components
+
+Instant messaging is separated into two parts:
+
+* Message storage
+* Delivery
+
+These core components allow users to join your chat at any time and view the message history and enable them to receive new messages in real time.
+
+### Message storage
+
+Most instant messaging platforms store messages to allow users to view the conversation history if messages were sent prior to their arrival. Part of the startup tasks when entering a chat room is loading said history and displaying it on screen. This can easily be done with Momento in two ways: with a **sorted set** or a **list** cache item, each with their own tradeoffs.
```

Comment:
After reading over this section, I'm not sure that I understand:
* why someone would choose to use sorted sets over lists; lists seem easier and you don't really describe a downside that I can see.
* why we should mention two options here rather than just picking one and being prescriptive about it?

If we wanna keep both data types in here then I think maybe you could stand to put a little more info about when to choose one over the other. but tbh I think if your focus is to teach people how to get up and running with instant messaging as quickly as possible then it might be simplest just to cover it with lists?

Reply:
totally possible that I am just missing something here though!

Reply:
Yeah that's a great point, i wanted to be complete but as you mention, this is for guidance. i'll take out sorted sets